tests/e2e-demos/multi-component.go

```diff
@@ -0,0 +1,161 @@
+				ApplicationName: "multi-component-application",
+				Components: []config.ComponentSpec{
+					{
+						Name:         "multi-component",
+						Type:         "public",
+						GitSourceUrl: "https://github.com/redhat-appstudio-qe/multi-component-example-main",
+					},
+				},
+			},
+		},
+	}
+
+	var compNameGo = testSpeicification.Tests[0].Components[0].Name + "-go"
+	var compNameNode = testSpeicification.Tests[0].Components[0].Name + "-nodejs"
+
+	BeforeAll(func() {
+		// Check to see if the github token was provided
+		Expect(utils.CheckIfEnvironmentExists(constants.GITHUB_TOKEN_ENV)).Should(BeTrue(), "%s environment variable is not set", constants.GITHUB_TOKEN_ENV)
+		// Check if 'has-github-token' is present, unless SKIP_HAS_SECRET_CHECK env var is set
+		if !utils.CheckIfEnvironmentExists(constants.SKIP_HAS_SECRET_CHECK_ENV) {
+			_, err := fw.HasController.KubeInterface().CoreV1().Secrets(RedHatAppStudioApplicationNamespace).Get(context.TODO(), ApplicationServiceGHTokenSecrName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Error checking 'has-github-token' secret %s", err)
+		}
+
+		_, err := fw.CommonController.CreateTestNamespace(AppStudioE2EApplicationsNamespace)
+		Expect(err).NotTo(HaveOccurred(), "Error when creating/updating '%s' namespace: %v", AppStudioE2EApplicationsNamespace, err)
+	})
+
+	// Remove all resources created by the tests
+	AfterAll(func() {
+
+	})
+
+	It("Create Red Hat AppStudio Application", func() {
+		createdApplication, err := fw.HasController.CreateHasApplication(testSpeicification.Tests[0].ApplicationName, AppStudioE2EApplicationsNamespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(createdApplication.Spec.DisplayName).To(Equal(testSpeicification.Tests[0].ApplicationName))
+		Expect(createdApplication.Namespace).To(Equal(AppStudioE2EApplicationsNamespace))
+	})
+
+	It("Check Red Hat AppStudio Application health", func() {
+		Eventually(func() string {
+			application, err = fw.HasController.GetHasApplication(testSpeicification.Tests[0].ApplicationName, AppStudioE2EApplicationsNamespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			return application.Status.Devfile
+		}, 3*time.Minute, 100*time.Millisecond).Should(Not(BeEmpty()), "Error creating gitOps repository")
+
+		Eventually(func() bool {
+			// application info should be stored even after deleting the application in application variable
+			gitOpsRepository := utils.ObtainGitOpsRepositoryName(application.Status.Devfile)
+
+			return fw.CommonController.Github.CheckIfRepositoryExist(gitOpsRepository)
+		}, 1*time.Minute, 1*time.Second).Should(BeTrue(), "Has controller didn't create gitops repository")
+	})
+
+	It("Create Red Hat AppStudio ComponentDetectionQuery for Component repository", func() {
+		cdq, err := fw.HasController.CreateComponentDetectionQuery(testSpeicification.Tests[0].Components[0].Name, AppStudioE2EApplicationsNamespace, testSpeicification.Tests[0].Components[0].GitSourceUrl, "", false)
```

Comment:
What if [0] don't exists

Reply:
I've added a length check in the BeforeAll block. Thanks!